internal/workflow/replicator.go

```diff
@@ -176,6 +193,7 @@ func (w *Replicator) execute(ctx workflow.Context, request *ReplicatorRequest) e
 								zap.Error(err),
 							)
 						}
+						responses = append(responses, *replicatorResponse)
```

Comment:
appending an value to a slice is not thread safe, and not safe from multiple workflow activities.  You'll need to create a workflow.NewNamedBufferedChannel, and use the send and ReceivdAsync to send the response across the workflows.

Reply:
Sure. I will try to use the NewNamedBufferedChannel.